core/scratch_block_comment.js

```diff
@@ -0,0 +1,404 @@
+    // No change.
+    return;
+  }
+  Blockly.Events.fire(
+      new Blockly.Events.Ui(this.block_, 'commentOpen', !visible, visible));
+  if ((!this.block_.isEditable() && !this.textarea_) || goog.userAgent.IE) {
+    // Steal the code from warnings to make an uneditable text bubble.
+    // MSIE does not support foreignobject; textareas are impossible.
+    // http://msdn.microsoft.com/en-us/library/hh834675%28v=vs.85%29.aspx
+    // Always treat comments in IE as uneditable.
+    Blockly.Warning.prototype.setVisible.call(this, visible);
+    return;
+  }
+  // Save the bubble stats before the visibility switch.
+  var text = this.getText();
+  var size = this.getBubbleSize();
+  if (visible) {
+    // Decide on placement of the bubble based on knowledge of the block
+    // that owns this comment:
+    if (this.isMinimized_) {
+      // If the block is minimized, ignore the given x and y
+      this.x = this.iconXY_.x + 16;
+      this.y = this.iconXY_.y - (Blockly.ScratchBubble.TOP_BAR_HEIGHT / 2);
+    } else if (!this.x && this.x != 0 && !this.y && this.y != 0) {
+      // Check if the width of this block (and all it's children/descendents) is the
+      // same as the width of just this block
+      var fullStackWidth = Math.floor(this.block_.getHeightWidth().width);
+      var thisBlockWidth = Math.floor(this.block_.svgPath_.getBBox().width);
+      if (fullStackWidth == thisBlockWidth && !this.block_.parentBlock_) {
+        this.x = this.iconXY_.x + 32;
+      } else {
+        this.x = this.iconXY_.x + fullStackWidth + 32;
+      }
+      this.y = this.iconXY_.y - (Blockly.ScratchBubble.TOP_BAR_HEIGHT / 2);
+    }
+
+    // Create the bubble.
+    this.bubble_ = new Blockly.ScratchBubble(
+        /** @type {!Blockly.WorkspaceSvg} */ (this.block_.workspace),
+        this.createEditor_(), this.iconXY_, this.width_, this.height_,
+        this.x, this.y, this.isMinimized_);
+    this.bubble_.setAutoLayout(false);
+    this.bubble_.registerResizeEvent(this.resizeBubble_.bind(this));
+    this.bubble_.registerMinimizeToggleEvent(this.toggleMinimize_.bind(this));
+    this.bubble_.registerDeleteEvent(this.dispose.bind(this));
+    this.bubble_.registerContextMenuCallback(this.showContextMenu_.bind(this));
+    this.updateColour();
+  } else {
+    // Dispose of the bubble.
+    this.bubble_.dispose();
+    this.bubble_ = null;
+    this.textarea_ = null;
+    this.foreignObject_ = null;
+    this.label_ = null;
+  }
+  // Restore the bubble stats after the visibility switch.
+  this.setText(text);
+  this.setBubbleSize(size.width, size.height);
+};
+
+/**
+ * Toggle the minimization state of this comment.
+ * @private
+ */
+Blockly.ScratchBlockComment.prototype.toggleMinimize_ = function() {
+  this.setMinimized(!this.isMinimized_);
+};
+
+/**
+ * Set the minimized state for this comment.
+ * @param {boolean} minimize Whether the comment should be minimized
+ */
+Blockly.ScratchBlockComment.prototype.setMinimized = function(minimize) {
```

Comment:
You have setMinimized defined here and in ScratchBubble.  Is the version in ScratchBubble ever used?

Reply:
The version in ScratchBubble is used inside this function e.g.: `this.bubble_.setMinimized(true, this.getLabelText());`